### PR TITLE
DCS-813 Ignore current booking when calculating availability during update

### PR DESCRIPTION
**Pull request by andrewrlee**
The rooms that are occupied by the booking that is being updated shouldn't be removed from the list of available rooms

### CODE REVIEW

File: backend/routes/amendBooking/changeDateAndTimeController.ts [hunk 1]
```diff
@@ -57,18 +57,15 @@ export default class ChangeDateAndTimeController {
...
-        res.locals,
-        {
-          agencyId,
-          date: form.date,
-          startTime: form.startTime,
-          endTime: form.endTime,
-          preRequired: form.preAppointmentRequired,
-          postRequired: form.postAppointmentRequired,
-        },
-        parseInt(bookingId, 10)
```
Comment (andrewrlee) [inline backend/routes/amendBooking/changeDateAndTimeController.ts:70]: no need for video booking id to be a separate field - the update object can hold all fields

File: backend/routes/amendBooking/selectAvailableRoomsController.ts [hunk 2]
```diff
@@ -66,6 +63,7 @@ export default class SelectAvailableRoomsController {
 
       const { isAvailable } = await this.availabilityCheckService.getAvailability(res.locals, {
         agencyId: bookingDetails.agencyId,
+        videoBookingId: parseInt(bookingId, 10),
```
Comment (andrewrlee) [inline backend/routes/amendBooking/selectAvailableRoomsController.ts:66]: This is the main reason for change - were completely ignoring the booking on the final availability check

File: integration-tests/integration/bookings/amendBooking/amendBooking.spec.js [hunk 3]
```diff
@@ -235,6 +214,38 @@ context('A user can amend a booking', () => {
 
     CourtVideoLinkBookingsPage.verifyOnPage()
```
Comment (andrewrlee) [inline integration-tests/integration/bookings/amendBooking/amendBooking.spec.js:216]: by the end of the test, the availability check will have been made 3 times

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**